community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreStatement.java

```diff
@@ -58,18 +58,19 @@
     private final NeoStores neoStores;
     private final NodeStore nodeStore;
     private final RelationshipStore relationshipStore;
-    private final IndexReaderFactory indexReaderFactory;
-    private final Supplier<LabelScanReader> labelScanReaderSupplier;
+    private final Supplier<IndexReaderFactory> indexReaderFactorySupplier;
+    private IndexReaderFactory indexReaderFactory;
+    private final Supplier<LabelScanReader> labelScanStore;
     private LabelScanReader labelScanReader;
     private boolean closed;
 
     public StoreStatement( final NeoStores neoStores, final LockService lockService,
-            IndexReaderFactory indexReaderFactory,
+            Supplier<IndexReaderFactory> indexReaderFactory,
```

Comment:
Maybe it is better to pass in `IndexingService` and let `StoreStatement` create `IndexReaderFactory` on it's own. Feels like passing in a factory gives no benefits.